src/server.js

```diff
@@ -449,6 +452,7 @@ export class Server extends EventEmitter {
                 reject(err);
             };
             const onListening = () => {
+                this.port = this.server.address().port;
```

Comment:
Can we update some existing unit test to check that `this.port` remains set correctly to the previous value, if it was null, undefined or 0?

Reply:
not sure, i added separate test for the first time